packages/gatsby/src/schema/extensions/index.js

```diff
@@ -0,0 +1,210 @@
+const {
+  GraphQLBoolean,
+  GraphQLNonNull,
+  GraphQLDirective,
+  GraphQLString,
+  DirectiveLocation,
+  defaultFieldResolver,
+} = require(`graphql`)
+const { GraphQLJSON } = require(`graphql-compose`)
+const report = require(`gatsby-cli/lib/reporter`)
+
+const { link, fileByPath } = require(`../resolvers`)
+const { getDateResolver } = require(`../types/date`)
+
+// Reserved for internal use
+const internalExtensionNames = [
+  `addDefaultResolvers`,
+  `createdFrom`,
+  `directives`,
+  `infer`,
+  `plugin`,
+]
+
+const typeExtensions = {
```

Comment:
This global state should be in Redux and passed from the top like for other global state.

Reply:
Agreed